batch/batch_test.go

```diff
@@ -300,3 +300,28 @@ func TestError(t *testing.T) {
 	}
 	wg.Wait()
 }
+
+func TestNoWithBatching(t *testing.T) {
+	f := (&batch.Func{
+		Many: func(ctx context.Context, args []interface{}) ([]interface{}, error) {
+			return nil, nil
+		},
+	}).Invoke
+
+	ctx := context.Background()
+
+	var wg sync.WaitGroup
+	for i := 0; i < 20; i++ {
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			defer func() {
+				if r := recover(); r != "WithBatching must be called on the context before using Func" {
+					t.Errorf("Incorrect panic message %s for %d", r, i)
+				}
+			}()
+			f(ctx, i)
```

Comment:
consider `assert.PanicsWithError` instead. if this didn't panic at all (i.e. without your fix), i think your test would still pass.

Reply:
Ohh that's a much cleaner way to write it. Fixed